lib/AST/ASTImporter.cpp

```diff
@@ -4397,8 +4397,22 @@ Decl *ASTNodeImporter::VisitClassTemplateDecl(ClassTemplateDecl *D) {
           // We found a forward declaration but the class to be imported has a
           // definition.
           if (D->isThisDeclarationADefinition() &&
-              !FoundTemplate->isThisDeclarationADefinition())
-            continue;
+              !FoundTemplate->isThisDeclarationADefinition()) {
+
+            // Lookup will not find the definition if the fwd decl is more
+            // recent. FIXME Is this a lookup error? Should we fix
+            // localUncachedLookup?
+            // So, lets try to get the definition if available.
+            if (CXXRecordDecl *ToDefinition =
```

Comment:
Now I think, this block should go before the structural Eq check.
Because we should compare the two definitions not a definition and a fwd decl!